src/corecel/math/detail/SoftEqualTraits.hh

```diff
@@ -44,7 +44,7 @@ struct SoftEqualTraits<double>
 {
     using value_type = double;
     static CELER_CONSTEXPR_FUNCTION value_type sqrt_prec() { return 1.0e-6; }
-    static CELER_CONSTEXPR_FUNCTION value_type rel_prec() { return 1.0e-12; }
+    static CELER_CONSTEXPR_FUNCTION value_type rel_prec() { return 1.0e-8; }
```

Comment:
Oh dear this is far too wide ranging a change. We need to drill down to this one test. It looks like the failing tests may be in a part of the solver that's numerically unstable or suffers from catastrophic rounding or something, because it's producing very different answers for single precision. I saw something similar in the nuclear form factor PR: https://github.com/celeritas-project/celeritas/pull/1398/commits/664ac3847ca47e6c4fbff86e6dbe3774ef87a277 .

The best solution here might just be to change `EXPECT_SOFT_EQ` to `EXPECT_SOFT_NEAR` to add the third `tolerance` option (and make that one 1e-8).

Reply:
I agree the tolerance would be too wide. For now, let's use `EXPECT_SOFT_NEAR`.